### PR TITLE
[BV] Add sending email

### PR DESCRIPTION
**Pull request by arsentur**
**Description:**
- Add `email_notifications` setting which allows sending email to the learner who earned a grade.
- Add helper method which sends emails.

**YouTrack:**
[BV-150](https://youtrack.raccoongang.com/issue/BV-150)

**Post Merge:**
- [ ] Squash commits
- [ ] Delete branch
- [ ] Add related-project tag

### CODE REVIEW

File: lti_consumer/lti_consumer.py [hunk 1]
```diff
@@ -422,12 +422,21 @@ class LtiConsumerXBlock(StudioEditableXBlockMixin, XBlock):
         scope=Scope.settings
     )
 
+    email_notifications = Boolean(
+        display_name=_("Enable Email Notification"),
+        help=_(
+            "Enables email notification when a student earned a grade."
```
Comment (idegtiarov) [inline lti_consumer/lti_consumer.py]: Spell: ```Send email notification when student earns a grade.```

File: lti_consumer/outcomes.py [hunk 2]
```diff
@@ -188,6 +192,28 @@ def handle_request(self, request):
             failure_values['imsx_description'] = "User not found."
             return response_xml_template.format(**failure_values)
 
+        # Sends an email when the user earned a grade if xblock setting "email_notifications" is enabled.
```
Comment (idegtiarov) [inline lti_consumer/outcomes.py]: ditto: s/earned/earns/

File: lti_consumer/templates/email/user_score_notification.html [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                        <td width="70">
+                            <a href="${lms_root}"><img src="${site_logo}" width="90" height="90" alt="Go to Your ${platform_name} Home Page"/></a>
+                        </td>
+                    </tr>
+                    <tr>
+                        <td>
+                            <h1>${_("Review available")}</h1>
+                            <p style="color: rgba(0,0,0,.75);">
+                                % if user_first_name:
+                                Dear ${user_first_name}
```
Comment (idegtiarov) [inline lti_consumer/templates/email/user_score_notification.html]: I would use condition case only for sensitive part like `${user_first_name}`

File: lti_consumer/outcomes.py [hunk 4]
```diff
@@ -203,3 +229,20 @@ def handle_request(self, request):
         unsupported_values['imsx_messageIdentifier'] = escape(imsx_message_identifier)
         log.debug("[LTI]: Incorrect action.")
         return response_xml_template.format(**unsupported_values)
+
+    def send_email(self, to_addr, context):
+        """
+        Helper function which sends an email with needed context.
```
Comment (idegtiarov) [inline lti_consumer/outcomes.py]: typo: s/which sends an email with needed context./which sends an email with required context./

File: lti_consumer/templates/email/user_score_notification.txt [hunk 5]
```diff
@@ -0,0 +1,16 @@
+<%
+from django.utils.translation import ugettext as _
+%>
+
+${_("Review available")}
+
+% if user_first_name:
+Dear ${user_first_name}
```
Comment (idegtiarov) [inline lti_consumer/templates/email/user_score_notification.txt]: ditto

**Review by idegtiarov — CHANGES_REQUESTED** [review]
Some minor nits are mentioned inline, other looks good to me.

File: lti_consumer/lti_consumer.py (hunk 1 above)
Comment (arsentur) [reply lti_consumer/lti_consumer.py]: Done

File: lti_consumer/outcomes.py (hunk 2 above)
Comment (arsentur) [reply lti_consumer/outcomes.py]: Done

File: lti_consumer/templates/email/user_score_notification.html (hunk 3 above)
Comment (arsentur) [reply lti_consumer/templates/email/user_score_notification.html]: Done

File: lti_consumer/templates/email/user_score_notification.txt (hunk 5 above)
Comment (arsentur) [reply lti_consumer/templates/email/user_score_notification.txt]: Done

File: lti_consumer/outcomes.py [hunk 6]
```diff
@@ -188,6 +192,28 @@ def handle_request(self, request):
...
+                "xblock_url": settings.LMS_ROOT_URL + reverse(
+                    'jump_to', args=[self.xblock.course_id, self.xblock.location]
+                ),
+                "site_logo": '{root_url}/static/{site_theme}/images/lti_email_logo.png'.format(
+                    root_url=settings.LMS_ROOT_URL,
+                    site_theme=settings.DEFAULT_SITE_THEME,
+                ),
+            }
+
+            self.send_email(real_user.email, context)
```
Comment (cmltaWt0) [inline lti_consumer/outcomes.py]: @arsentur 
Have you considered to use a background task for this?

Comment (NikolayBorovenskiy) [reply lti_consumer/outcomes.py]: @cmltaWt0 @arsentur is too busy now and I as a free option have implemented your remark.  Thanks for it and please do a finale reviews I bet.

**Comment by NikolayBorovenskiy** [comment]
@idegtiarov @cmltaWt0 we have coped together with @arsentur with your review comments and are waiting for the final one. Thanks in advance.

**Review by idegtiarov — APPROVED** [review]
LGTM!

**Pull request merged**